.github/workflows/canister-tests.yml

```diff
@@ -344,27 +344,48 @@ jobs:
           path: .
 
       - name: Deploy II and run tests
-        working-directory: demos/using-dev-build
         run: |
           set -euo pipefail
 
+          # Copy example to make sure it does not rely on living inside the II repo
+          builddir=$(mktemp -d)
+          cp -r ./demos/using-dev-build/. "$builddir"
+          pushd "$builddir"
+
+          # Install npm deps
           npm ci
 
-          # Create a fake curl so that we use this build's wasm; also create a "witness"
-          # file to make sure the fake curl was run (rm $witness will fail otherwise)
-          curl_dir=$(mktemp -d); witness=$(mktemp)
+          # Create a fake curl so that we use this build's wasm and did files; also create "witness"
+          # files to make sure the fake curl was run (rm $witness_foo will fail otherwise)
+          curl_dir=$(mktemp -d); witness_wasm=$(mktemp)
```

Comment:
How does this witness mechanism work exactly? `witness_wasm=$(mktemp)` creates the file already, so having executed this line `rm $witness_wasm` will not fail, even if the `touch $witness_wasm` is not executed. Or am I missing something?
Also you are creating two `curl_dir`s. This is not necessary, right?

Reply:
That's absolutely correct! Good catch